ts/packages/ts/src/commands/UpsertCommand.ts

```diff
@@ -5,65 +5,96 @@ import { ConfigError } from '../utils';
     super(parameters);
   }
 
-  async run() {
-    const { store, set, schema, configs } = this.parameters;
+  private validateConfigValue({ key, value, isFromPipe }: { key: string; value: string; isFromPipe: boolean }) {
+    const { store, set, schema } = this.parameters;
 
-    await store.init();
+    const errorScope: [string, string][] = [
+      ['UpsertCommand', `store:${store.type};set:${set.path};schema:${schema.name}`],
+      ['parameters.configs', `key:${key};value:${value}`],
+    ];
 
-    const upsertConfigs = _(configs)
-      .entries()
-      .map<Config>(([key, value]) => {
-        const errorScope: [string, string][] = [
-          ['UpsertCommand', `store:${store.type};set:${set.path};schema:${schema.name}`],
-          ['parameters.configs', `key:${key};value:${value}`],
-        ];
+    const cfgu = schema.contents[key];
 
-        const cfgu = schema.contents[key];
+    if (!isFromPipe) {
+      if (!cfgu) {
+        throw new ConfigError(
+          'invalid config key',
+          `key "${key}" must be declared on schema ${schema.name}`,
+          errorScope,
+        );
+      }
+      if (value && cfgu?.template) {
+        throw new ConfigError('invalid config value', `keys declared with template mustn't have a value`, errorScope);
+      }
+    }
 
-        if (!cfgu) {
-          throw new ConfigError(
-            'invalid config key',
-            `key "${key}" must be declared on schema ${schema.name}`,
-            errorScope,
-          );
-        }
+    if (value && cfgu) {
+      try {
+        ConfigSchema.CFGU.VALIDATORS.valueOptions(cfgu, value);
+        ConfigSchema.CFGU.VALIDATORS.valueType(cfgu, value);
+      } catch (error) {
+        throw error?.appendScope?.(errorScope) ?? error;
+      }
+    }
+  }
+
+  async run() {
+    const { store, set, schema, configs, pipe } = this.parameters;
 
-        if (value) {
-          if (cfgu.template) {
-            throw new ConfigError(
-              'invalid config value',
-              `keys declared with template mustn't have a value`,
-              errorScope,
-            );
-          }
+    if (_.isEmpty(configs) && _.isEmpty(pipe)) {
+      return;
+    }
+
+    await store.init();
 
-          try {
-            ConfigSchema.CFGU.VALIDATORS.valueOptions(cfgu, value);
-            ConfigSchema.CFGU.VALIDATORS.valueType(cfgu, value);
-          } catch (error) {
-            throw error?.appendScope?.(errorScope) ?? error;
-          }
+    const pipeConfigs = _(pipe)
+      .entries()
+      .filter(([key]) => {
+        const schemaCfgu = schema.contents[key];
+        // * Filters keys from pipe that are not declared on the schema
+        if (!schemaCfgu) {
+          return false;
         }
+        // * Filters keys that are declared as a template in the provided schema
+        return !schemaCfgu.template;
+      })
+      .map(([key, value]) => {
+        this.validateConfigValue({ key, value: value.result.value, isFromPipe: true });
```

Comment:
If this is here for extra-validation, ok. If we assume pipe is from EvalResult this is redundant.

Reply:
Extra validation, we want to ensure that the eval result matches the cfgu declarations of the provided schema.